Gems/Camera/Code/Source/EditorCameraComponent.cpp

```diff
@@ -193,6 +196,18 @@ namespace Camera
         return AZ::Edit::PropertyRefreshLevels::AttributesAndValues;
     }
 
+    AZ::Crc32 EditorCameraComponent::AlignCameraWithViewClicked()
+    {
+        bool result = false;
+        AZ::Transform transform = AZ::Transform::CreateIdentity();
+        EditorCameraRequests::Bus::BroadcastResult(result, &EditorCameraRequests::GetActiveCameraTM, transform);
+        if (result)
+        {
+            AZ::TransformBus::Event(GetEntityId(), &AZ::TransformInterface::SetWorldTM, transform);
+        }
+        return AZ::Edit::PropertyRefreshLevels::AttributesAndValues;
+    }
+
```

Comment:
Should this have an undo batch?

Reply:
I think I understand. This is doing two things at once, align the camera and possess the camera. 

The way it is described, I think it should start the undo batch, mark the entity dirty, update the transform, but not call SetViewFromEntityPerspective. There is already a button and behavior that manages possession or “becoming this camera”. 

Even though you disable the button in the edit context, the function should also fire a warning and return early if IsThisCamera returns true, in case someone calls it from script or in a different context.

Lastly, if you need custom undo/redo behavior that's not tied to entity state, you can create a derived URSequencePoint class, and add an instance of it as a child of the undo batch. I was searching and I am surprised that there's no simple URSequencePoint class that takes in a couple of lambdas.